File: JMMinistry.Common/Dtos/Discipleship/DiscipleshipNoteEntryDto.cs
namespace JMMinistry.Common.Dtos.Discipleship
{
    public class DiscipleshipNoteEntryDto
    {
        public int Id { get; set; }
        public required string Content { get; set; }
        public DateTime Date { get; set; }
        public DateTime CreatedAt { get; set; }
        public int NoteId { get; set; }
        public required string AuthorId { get; set; }
    }
}

File: the_vine_hub_app/lib/core/network/api/discipleship/models/discipleship_note_entry_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'discipleship_note_entry_dto.freezed.dart';
part 'discipleship_note_entry_dto.g.dart';

@freezed
abstract class DiscipleshipNoteEntryDto with _$DiscipleshipNoteEntryDto {
  const factory DiscipleshipNoteEntryDto({
    required String id,
    String? content,
    required DateTime date,
    required DateTime createdAt,
    required String noteId,
    String? authorId,
  }) = _DiscipleshipNoteEntryDto;

  factory DiscipleshipNoteEntryDto.fromJson(Map<String, dynamic> json) =>
      _$DiscipleshipNoteEntryDtoFromJson(json);
}
